### PR TITLE
Add #empty? to Tempfile, StringIO, File::Stat

### PR DESCRIPTION
**Pull request by mikegee**
Rubocop prefers `empty?` over `length == 0` and `size == 0`, which is great for String, Array, Hash, etc. It would be nice if more classes implemented `#empty?` for consistency.

See related discussion at https://github.com/bbatsov/rubocop/issues/2841

https://bugs.ruby-lang.org/issues/14136

### CODE REVIEW

**Comment by mikegee** [comment]
I couldn't find where to add tests for `File::Stat#empty?`, but that one just an alias. Maybe it isn't valuable enough to test that one. ¯\\\_(ツ)_/¯

File: ext/stringio/stringio.c [hunk 1]
```diff
@@ -1523,6 +1523,24 @@ strio_size(VALUE self)
...
+ */
+static VALUE
+strio_empty(VALUE self)
+{
+    VALUE string = StringIO(self)->string;
+    if (NIL_P(string)) {
+      rb_raise(rb_eIOError, "not opened");
+    }
+    long len = RSTRING_LEN(string);
+    if (len == 0) { return Qtrue; }
```
Comment (esparta) [inline ext/stringio/stringio.c]: Since `len` is not used anywhere else but that `if`, can this be equivalent?
```c
if (RSTRING_LEN(string) == 0) { return Qtrue; }
```

Comment (mikegee) [reply ext/stringio/stringio.c]: Good point!

**Comment by Drenmi** [comment]
Would love to see this merged in the name of 🦆 typing. 🙂

**Comment by bbatsov** [comment]
Same here.
